requirements/edx/base.txt

```diff
@@ -91,6 +91,7 @@ django-ratelimit-backend==0.6
 unicodecsv==0.9.4
 django-require==1.0.6
 pyuca==1.1
+wrapt==1.10.5
```

Comment:
Are there any performance implications of using wrapt instead of
setuptools?  Also are you aware that we recently updated to the latest
version of setuptools?(Until Thursday we were on a really old version of
distribute). Do the problem still exist in the new version of setuptools?
On Jul 10, 2015 1:13 PM, "Utkarsh" <notifications@github.com> wrote:

> In requirements/edx/base.txt
> <https://github.com/edx/edx-platform/pull/8835#discussion_r34375061>:
>
> > @@ -91,6 +91,7 @@ django-ratelimit-backend==0.6
> >  unicodecsv==0.9.4
> >  django-require==1.0.6
> >  pyuca==1.1
> > +wrapt==1.10.5
>
> @feanil <https://github.com/feanil> --
> About library- the wrapt is used to avoid conflicts between @contract and
> any parameterized decorator. setuptools.wraps() fails to do so, but
> @wrapt.decorator does it perfectly.
> IMO, wrapt is better than setuptools.wraps().
>
> Is it going to be used on production?
>
>    - Yes. The library Call Stack Manager is used to get unique, unknown
>    call stacks of StudentModule and StudentModuleHistory on production.
>    Due to this, we would be able to make sure that all call to
>    studentmodule and studentmodulehistory will go through
>    edx-user-state-client interface.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/8835/files#r34375061>.
>